bin/merge_featurecounts.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env/python
+
+import argparse
+import os
+import re
+from collections import defaultdict
+
+
+def do_the_thing(dest_dir,out_file,input_files):
+   table_dict=defaultdict(dict)
+   sample_names=[]
+   gene_list=[]
+   for input_file in input_files:
+       if not gene_list:
+           first_file=True
+       print"Reading from {}".format(input_file)
+       sample_name=os.path.basename(input_file)
+       sample_name= sample_name.replace("Aligned.sortedByCoord.out_gene.featureCounts.txt","")
+       sample_names.append(sample_name)
+       table_dict[sample_name]=dict()
+     
+       #Extract the gene list from the first file
+       if first_file:
+           with open(input_file, 'r') as f:
+               for line in f:
+                   if not line.startswith('E'):
+                       continue
+                   line_info=line.split('\t')
+                   gene=line_info[0]
+                   gene_list.append(gene)
+       #Read the rest of the files
+       with open(input_file, 'r') as f:
+           for line in f:
+               if not line.startswith('E'):
+                   continue
+               line_info=line.split('\t')
+               gene=line_info[0]
```

Comment:
In fact, if `gene_list` is a `set` instead of a `list` you can just do it for every file and you don't need to track whether you're in the `first_file` or not.

```python
gene_list.update(gene)
```